tests/drivers/flash/interface_test/src/main.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/random/random.h>
+
+#ifdef CONFIG_BOOTLOADER_MCUBOOT
+#define TEST_FLASH_PART_NODE DT_NODELABEL(boot_partition)
+#else
+#define TEST_FLASH_PART_NODE DT_NODELABEL(slot1_partition)
+#endif
+
+#define TEST_FLASH_PART_OFFSET DT_REG_ADDR(TEST_FLASH_PART_NODE)
+#define TEST_FLASH_PART_SIZE   DT_REG_SIZE(TEST_FLASH_PART_NODE)
+
+#define TEST_FLASH_CONTROLLER_NODE DT_MTD_FROM_FIXED_PARTITION(TEST_FLASH_PART_NODE)
+
+#define PATTERN_SIZE 256
+
+static const struct device *flash_controller = DEVICE_DT_GET(TEST_FLASH_CONTROLLER_NODE);
+
+static uint8_t pattern[PATTERN_SIZE];
+
+static const off_t test_area_offset = TEST_FLASH_PART_OFFSET;
+static const size_t test_area_size = TEST_FLASH_PART_SIZE & ~0xff;
+static const off_t test_area_end = test_area_offset + test_area_size;
+
+static int verify_block(off_t pos, uint8_t *expected_data, size_t size)
+{
+	uint8_t buffer[size];
+
+	if (flash_read(flash_controller, pos, buffer, size) != 0) {
+		return -EIO;
+	}
+
+	if (memcmp(expected_data, buffer, size) != 0) {
+		return -EBADMSG;
+	}
+
+	return 0;
+}
+
+static void *flash_setup(void)
+{
+	bool pattern_available = true;
+
+	printf("test_area_size = %u MB\n", test_area_size >> 20);
+
+	/* Initialize pattern */
+	for (int i = 0; i < sizeof(pattern); i++) {
+		pattern[i] = i;
+	}
+
+	for (off_t pos = test_area_offset; pos < test_area_end; pos += PATTERN_SIZE) {
+		int res = verify_block(pos, pattern, sizeof(pattern));
+
+		if (res == -EBADMSG) {
+			pattern_available = false;
+			break;
+		}
+
+		/* fail at any other errors */
+		zassert_ok(res);
```

Comment:
I am not sure about this sequence.
When you get pattern across the entire partition, then you will get some N * zassert_ok results from this block; but if you do not get these, the output here is cut short. This would look like the test is unstable, or does not seem to follow the same paths. I think that here should be single check of failure, and there is no place for zassert_ok in the loop.

Reply:
You can use `zexpect_ok` to continue afterwards.